fish_prompt.fish

```diff
@@ -81,9 +81,8 @@ function fish_prompt
     end
 
     # Check if there is an upstream configured
-    set -l has_upstream (command git rev-parse --abbrev-ref @'{u}' ^/dev/null)
-
-    if test -n "$has_upstream"
+    command git rev-parse --abbrev-ref '@{upstream}' >/dev/null ^&1
```

Comment:
* is the subcommand problematic here?
* what is the `'@{upstream}'` doing? My 

       --abbrev-ref[=(strict|loose)]
           A non-ambiguous short name of the objects name. The option core.warnAmbiguousRefs is used to select the strict abbreviation mode.